store/tikv/region_cache.go

```diff
@@ -491,11 +440,7 @@ func (c *RegionCache) OnSendFail(bo *Backoffer, ctx *RPCContext, scheduleReload
 	tikvRegionCacheCounterWithSendFail.Inc()
 	r := c.getCachedRegionWithRLock(ctx.Region)
 	if r != nil {
-		if ctx.Store.storeType == kv.TiKV {
-			c.switchNextPeer(r, ctx.PeerIdx, err)
-		} else {
-			c.switchNextFlashPeer(r, ctx.PeerIdx, err)
-		}
+		c.switchPeerForNextRetry(ctx.Store.storeType, ctx.NextOnFail, r, ctx.PeerIdx, err)
```

Comment:
`switchPeerForNextRetry` will change `currIdx`, so I guess it must be called for sending to leader?

Reply:
It should be used for sending to TiKV leader or sending to tiflash, but follower read seems shouldn't call this.

Reply:
I want to remove `seed` from follower read, then follower read will share the same logic with tiflash. Is there any known problem?